libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+ *         #TRANSPORT_SOCKET_STATUS_DNS_FAILURE, #TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE on failure.
+ */
+static TransportSocketStatus_t establishConnect( NetworkContext_t * pNetworkContext,
+                                                 const ServerInfo_t * pServerInfo,
+                                                 const SocketsConfig_t * pSocketsConfig );
+
+/**
+ * @brief Set up TLS configurations for the socket.
+ *
+ * @param[in] pSocketsConfig The session configuration.
+ * @param[in] tcpSocket An initialized socket to secure.
+ * @param[in] pHostName Remote server name for SNI.
+ * @param[in] hostnameLength The length of `pHostName`.
+ *
+ * @return #SOCKETS_ERROR_NONE or #SOCKETS_SOCKET_ERROR.
+ */
+static int32_t tlsSetup( const SocketsConfig_t * pSocketsConfig,
+                         Socket_t tcpSocket,
+                         const uint8_t * pHostName,
+                         size_t hostnameLength );
+
+
+/*-----------------------------------------------------------*/
+
+int32_t SecureSocketsTransport_Send( NetworkContext_t * pNetworkContext,
+                                     const void * pMessage,
+                                     size_t bytesToSend )
+{
+    size_t bytesSent = 0;
+    int32_t TransportSocketStatus = SOCKETS_ERROR_NONE;
+    Socket_t xSocket = SOCKETS_INVALID_SOCKET;
+
+    if( ( pMessage == NULL ) ||
+        ( bytesToSend <= ( size_t ) 0 ) ||
+        ( pNetworkContext == NULL ) ||
+        ( pNetworkContext->pContext == NULL ) )
+    {
+        LogError( ( "TransportSendSecureSockets bad parameters pMessage %p, bytesToSend %d,  pvContext %p",
+                    pMessage, bytesToSend, pNetworkContext->pContext ) );
+        bytesSent = ( size_t ) SOCKETS_EINVAL;
+    }
+    else
+    {
+        xSocket = ( Socket_t ) pNetworkContext->pContext;
+        TransportSocketStatus = SOCKETS_Send( xSocket,
+                                              pMessage,
+                                              bytesToSend,
+                                              0 );
+
+        /* If an error occurred, a negative value is returned. @ref SocketsErrors. */
+        if( TransportSocketStatus >= ( int32_t ) 0 )
+        {
+            bytesSent = ( size_t ) TransportSocketStatus;
+            LogInfo( ( "Success %ld while sending data.", TransportSocketStatus ) );
+        }
+        else
+        {
+            LogError( ( "Error %ld while sending data.", TransportSocketStatus ) );
+            bytesSent = ( size_t ) SOCKETS_SOCKET_ERROR;
+        }
+    }
+
+    return ( int32_t ) bytesSent;
+}
+
+/*-----------------------------------------------------------*/
+
+int32_t SecureSocketsTransport_Recv( NetworkContext_t * pNetworkContext,
+                                     void * pBuffer,
+                                     size_t bytesToRecv )
+{
+    int32_t TransportSocketStatus = 0;
+    size_t bytesReceived = SOCKETS_SOCKET_ERROR;
```

Comment:
Should this be of `in32_t` type due to the return type of `SOCKETS_Recv`?

Reply:
fixed in 5fabf463be948c1821f30e186dcb9a13009ba915